queries/src/main/java/org/apache/geode/examples/queries/Example.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+import org.apache.geode.cache.query.Query;
+import org.apache.geode.cache.query.QueryService;
+import org.apache.geode.cache.query.SelectResults;
+
+
+public class Example implements Consumer<Region<Integer, EmployeeData>> {
+  public static void main(String[] args) {
+    // connect to the locator using default port 10334
+    ClientCache cache = new ClientCacheFactory().addPoolLocator("127.0.0.1", 10334)
+        .set("log-level", "WARN").create();
+
+    // create a local region that matches the server region
+    Region<Integer, EmployeeData> region =
+        cache.<Integer, EmployeeData>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+            .create("example-region");
+
+    Example e = new Example();
+    e.accept(region);
+    e.doQueries(cache, region);
+    cache.close();
+  }
+
+  @Override
+  public void accept(Region<Integer, EmployeeData> region) {
+    // put entries in the region
+    String[] firstNames =
+        "Alex,Bertie,Kris,Dale,Frankie,Jamie,Morgan,Pat,Ricky,Taylor,Casey,Jessie,Ryan,Skyler"
+            .split(",");
+    String[] lastNames =
+        "Able,Bell,Call,Driver,Forth,Jive,Minnow,Puts,Reliable,Tack,Catch,Jam,Redo,Skip".split(",");
+    int salaries[] = new int[] {60000, 80000, 75000, 90000, 100000};
+    int hours[] = new int[] {40, 40, 40, 40, 30, 20};
+    int emplNumber = 10000;
+    for (int index = 0; index < firstNames.length; index++) {
+      emplNumber = emplNumber + index;
+      Integer key = emplNumber;
+      String email = firstNames[index] + "." + lastNames[index] + "@example.com";
+      int salary = salaries[index % 5];
+      int hoursPerWeek = hours[index % 6];
+      EmployeeData value = new EmployeeData(firstNames[index], lastNames[index], emplNumber, email,
+          salary, hoursPerWeek);
+      region.put(key, value);
+    }
+
+    // count the values in the region
+    int inserted = region.keySetOnServer().size();
+    System.out.println(String.format("Counted %d keys in region %s", inserted, region.getName()));
+
+    // fetch and print all values in the region (without using a query)
+    region.keySetOnServer().forEach(key -> System.out.println(region.get(key)));
+
+  }
+
+  // Query using the API
+  private void doQueries(ClientCache cache, Region<Integer, EmployeeData> region) {
+    String regionName = region.getName();
+    QueryService queryService = cache.getQueryService();
+
+    // query for every entry in the region, and print query results
+    String queryString1 = "SELECT DISTINCT * FROM /" + regionName;
+    System.out.println(String.format("\nQuery: %s", queryString1));
+    Query query = queryService.newQuery(queryString1);
+    SelectResults<EmployeeData> results = null;
+    try {
+      results = (SelectResults<EmployeeData>) query.execute();
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
I think you should use a single `catch` block.  It simplifies the code and you probably want to abort if there's an error instead of trying to continue.